jsk_rosbag_tools/scripts/bag_to_video.py

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env python
+
+import argparse
+import datetime
+import os
+import sys
+
+import moviepy.editor as mp
+from moviepy.video.io.ffmpeg_writer import FFMPEG_VideoWriter
+
+from jsk_rosbag_tools import cv2
+from jsk_rosbag_tools.extract import extract_audio
+from jsk_rosbag_tools.extract import extract_image_topic
+from jsk_rosbag_tools.extract import get_image_topic_names
+
+
+def main():
+    parser = argparse.ArgumentParser(description='rosbag to video')
+    parser.add_argument('--out', '-o', default='',
+                        help='output directory path')
+    parser.add_argument('--fps', default=30,
+                        type=int)
+    parser.add_argument('--samplerate', '-r', type=int, help='sampling rate',
+                        default='16000')
+    parser.add_argument('--channels',
+                        type=int, default=1, help='number of input channels')
+    parser.add_argument('--audio-topic', type=str, default='/audio')
+    parser.add_argument('--image-topics', type=str,
+                        nargs='+', help='Topic name to extract.')
+    parser.add_argument('input_bagfile')
+    args = parser.parse_args()
+
+    input_bagfile = args.input_bagfile
+    if not os.path.exists(input_bagfile):
+        print('Input bagfile {} not exists.'.format(input_bagfile))
+        sys.exit(1)
+
+    if len(args.out) == 0:
+        basename = os.path.basename(input_bagfile)
+        dirname = os.path.dirname(input_bagfile)
+        filename, _ = os.path.splitext(basename)
+        out = os.path.join(dirname, filename)
+    else:
+        out = args.out
+    os.makedirs(out, mode=0o777, exist_ok=True)
+
+    fps = args.fps
+    dt = 1.0 / fps
+
+    wav_outpath = os.path.join(out, 'audio.wav')
+    audio_exists = extract_audio(input_bagfile, wav_outpath,
+                                 samplerate=args.samplerate,
+                                 channels=args.channels,
+                                 topic_name=args.audio_topic)
+
+    candidates_topic_names = get_image_topic_names(
+        input_bagfile, rgb_only=True)
+    if args.image_topics is None:
+        topic_names = candidates_topic_names
+    else:
+        topic_names = [tn for tn in args.image_topics
+                       if tn in candidates_topic_names]
+
+    for topic_name in topic_names:
+        images = extract_image_topic(input_bagfile, topic_name)
+
+        # remove 0 time stamp
+        stamp = 0.0
+        while stamp == 0.0:
+            stamp, _, img, _ = next(images)
+        start_stamp = stamp
+
+        if topic_name[0] == '/':
+            replaced_topic_name = topic_name[1:]
+        else:
+            replaced_topic_name = topic_name
+        replaced_topic_name = replaced_topic_name.replace('/', '--slash--')
+        output_videopath = os.path.join(
+            out, "{}.mp4".format(replaced_topic_name))
```

Comment:
How about adding an argument that specifies the name of the output file?

Reply:
Thanks. Modified